privatenetworks.go

```diff
@@ -0,0 +1,58 @@
+package scalingo
+
+import (
+	"context"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	"gopkg.in/errgo.v1"
+
+	httpclient "github.com/Scalingo/go-scalingo/v8/http"
+	"github.com/Scalingo/go-utils/pagination"
+)
+
+type PrivateNetworksService interface {
+	PrivateNetworksDomainsList(ctx context.Context, app string, page string, perPage string) (pagination.Paginated[[]PrivateNetworkDomain], error)
+}
+
+var _ PrivateNetworksService = (*Client)(nil)
+
+type PrivateNetworkDomain = string
+
+type PrivateNetworkDomainsRes struct {
+	Domains pagination.Paginated[[]PrivateNetworkDomain] `json:"domain_names"`
+}
+
+func (c *Client) PrivateNetworksDomainsList(ctx context.Context, app string, page string, perPage string) (pagination.Paginated[[]PrivateNetworkDomain], error) {
```

Comment:
issue: you should use `PaginationOpts` for the pagination. You can look at other places using this structure